### DISCUSSION TITLE
ideal video length that can be transcribed by whisper?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [openai/whisper#136] [Q&A] [answered]
i've read it processes videos in 30-second chunks.

my question is how long can the video be transcribed by whisper? 1-hour long, 8-hour long? should a video be used or audio for faster transcription?

i have tried https://huggingface.co/spaces/sensahin/YouWhisper on [a 15 mins video](https://www.youtube.com/watch?v=-TOYJHax5x8) & it accurately transcribed it into ~4117 words.

is there an upper limit? does the program `youwhisper` cut it into chunks 30-second chunks? i haven't used python in like 5 years so not totally sure about the code.

**Comment by jongwook** [comment, marked as answer]
The model natively supports 30-second inputs, and for audio inputs longer than that, we're using a set of (hacky) heuristics to perform transcription on sliding windows. The details are described in Section 4.5 of [the paper](https://cdn.openai.com/papers/whisper.pdf) and implemented in [`transcribe.py`](https://github.com/openai/whisper/blob/main/whisper/transcribe.py).

The script takes a number of manually tuned hyperparameters which works well on easy inputs but may need more adjustments for audio that is more difficult to transcribe or in a language that Whisper doesn't perform very well.

**Reply by FurkanGozukara** [reply]
I am waiting my graphic card to test it on Turkish. Do you have any plans to add translation from english to other languages?
